symfit/contrib/interactive_fit/interactive_fit.py

```diff
@@ -0,0 +1,194 @@
+        ncols = int(np.ceil(len(self._projections)/nrows))
+
+        self._plots = {}
+        for plotnr, proj in enumerate(self._projections, 1):
+            x, y = proj
+            ax = self.fig.add_subplot(ncols, nrows, plotnr,
+                                      label='{} {}'.format(x.name, y.name))
+            ax.set_title(ax.get_label())
+            ax.set_ylim(y_mins[y.name], y_maxs[y.name])
+            ax.set_xlim(x_mins[x.name], x_maxs[x.name])
+            # TODO reduce dimensionality.
+            ax.scatter(self.independent_data[x.name],
+                       self.dependent_data[y.name])
+
+            vals = self._get_data(x, y)
+            plot, = ax.plot(*vals, c='red')
+            self._plots[proj] = plot
+
+        i = 0.05
+        self._sliders = {}
+        for p in self.model.params:
+            if not p.fixed:
+                axbg = 'lightgoldenrodyellow'
+            else:
+                axbg = 'red'
+            # start-x, start-y, width, height
+            ax = self.fig.add_axes((0.162, i, 0.68, 0.03), axis_bgcolor=axbg, label=p.name)
+            val = p.value
+            if p.min is None:
+                minimum = 0
+            else:
+                minimum = p.min
+            if p.max is None:
+                maximum = 2 * val
+            else:
+                maximum = p.max
+
+            slid = plt.Slider(ax, p.name, minimum, maximum, valinit=val)
+            self._sliders[p] = slid
+            slid.on_changed(self._update_plot)
+            i += 0.05
+
+    def _update_plot(self, _):
+        """Callbak to redraw the plot to reflect the new parameter values."""
+        # Since all sliders call this same callback without saying who they are
+        # I need to update the values for all parameters. This can be
+        # circumvented by creating a seperate callback function for each
+        # parameter.
+        for p in self.model.params:
+            p.value = self._sliders[p].val
+        for indep_var, dep_var in self._projections:
+            plot = self._plots[(indep_var, dep_var)]
+            # TODO: reduce dimensionality of self._x_points and vals for this projection
+            vals = self._get_data(indep_var, dep_var)
+            plot.set_data(*vals)
+
+    def _get_data(self, independent_var, dependent_var):
+        """
+        Convenience method for evaluating the model, giving the projection
+        dependent_var, independent_var
+
+        Parameters
+        ----------
+        dependent_var : Variable
+            The dependent variable to calculate the data for
+        independent_var : Variable
+            The independent variable to calculate the data for
+
+        Returns
+        -------
+        x_points, y_points
+        """
+        x_points = self._x_points[independent_var.name]
+        arguments = {independent_var.name: x_points}
+        arguments.update({p.name: p.value for p in self.model.params})
```

Comment:
Because this happens so often, I build a nice method in support called ``key2str``. Check it out! It means you can build your dict with any type of object that has __str__ defined as keys, and upon calling you just do **key2str(my_dict)